src/core/Elsa.Core/Extensions/ElsaServiceCollectionExtensions.cs

```diff
@@ -59,9 +62,28 @@ public static class ElsaServiceCollectionExtensions
         }
 
         public static IServiceCollection AddActivity<T>(this IServiceCollection services) where T : class, IActivity =>
-            services
-                .AddTransient<T>()
-                .AddTransient<IActivity>(sp => sp.GetRequiredService<T>());
+            services.AddActivity(typeof(T));
```

Comment:
I'm wondering if we can't skip this registration altogether. After all, we can still instantiate activity types using `ActivatorUtilities` using `IServiceProvider` to resolve an activity's constructor arguments. Then we could even potentially register additional activity types at runtime.

Reply:
It is already possible now. I have only left the registration type as it is. The activities are no longer registered to DI, but to `ElsaOptions`. `ElsaOptions` is registered as an option and can be accessed anywhere. Thus, new activities can be added at any time.

Reply:
Great. Then there's no need to register the activity type as a service, right?

Reply:
Exactly.